atc/db/volume.go

```diff
@@ -351,17 +351,25 @@ func (volume *createdVolume) findWorkerBaseResourceTypeByBaseResourceTypeID(base
 	}, nil
 }
 
-func (volume *createdVolume) InitializeResourceCache(resourceCache UsedResourceCache) error {
+// InitializeResourceCache creates a worker resource cache and point current volume's
+// worker_resource_cache_id to the cache. When initialize a resource cache that is
+// generated on current worker, then sourceWorkerName can be empty.
```

Comment:
@aoldershaw I'm not following. Can you be more specific?

Reply:
@aoldershaw I intentionally keep two functions: `InitializeResourceCache` and `InitializeStreamedResourceCache`, because I consider they are logically different, one for initialize local generated resource cache, the other for initialize streamed-in resource cache. Keeping them separate helps make caller easier to decide how to use them.

With your comment, I think we should also make them different in module `db`, so I made the change in the last commit.